src/coreComponents/mesh/utilities/MeshMapUtilities.hpp

```diff
@@ -126,103 +131,167 @@ inline localIndex size1( ArrayOfSetsView< T const > const & map, localIndex cons
 //////////////////////////////////////////////////////////////////////////////////
 
 /**
- * @brief @return the value of the map
- * @tparam T type of map element
- * @param map reference to the map
- * @param i0 first-dimension index into the map
- * @param i1 second-dimension index into the map
+ * @brief Transposes an input map (array2d, ArrayOfArrays or ArrayOfSets)
+ * @tparam POLICY execution policy to use
+ * @tparam VIEW_TYPE type of view of the source map
+ * @param srcMap the source map
+ * @param dstSize number of target objects ("cols" of @p srcMap)
+ * @param overAlloc overallocation (extra capacity per row) for resulting map
+ * @return the transpose of @p srcMap stored as ArrayOfArrays (most general form)
  */
-template< typename T >
-GEOSX_HOST_DEVICE
-inline T const & value( arrayView1d< arrayView1d< T const > const > const & map, localIndex const i0, localIndex const i1 )
+template< typename POLICY, typename VIEW_TYPE >
+ArrayOfArrays< std::remove_const_t< typename VIEW_TYPE::ValueType > >
+transposeIndexMap( VIEW_TYPE const & srcMap,
```

Comment:
Nice work to put al these pieces of code in one place. Plus now it's way better documented too.